guarddog/reporters/sarif.py

```diff
@@ -188,3 +188,50 @@ def report_verify_sarif(package_path: str, rule_names: list[str], scan_results:
     runs = get_run(results, driver)
     log = get_sarif_log([runs])
     return json.dumps(log, indent=2)
+
+
+def report_scan_sarif(rule_names: list[str], scan_results: list[dict],
```

Comment:
can we add a unit test for it? it seems to be failing for a sample package I tried it on:


```
$ guarddog pypi scan first-exercise --output-format sarif
File "/Users/christophetd/workspace/guarddog/guarddog/cli.py", line 191, in _scan
    return_value = report_scan_sarif(list(ALL_RULES), results, ecosystem)
  File "/Users/christophetd/workspace/guarddog/guarddog/reporters/sarif.py", line 211, in report_scan_sarif
    "startLine": record["start"]["line"],
TypeError: string indices must be integers
```